deepchem/models/keras_model.py

```diff
@@ -460,28 +476,49 @@ def fit_generator(self,
 
       if checkpoint_interval > 0 and current_step % checkpoint_interval == checkpoint_interval - 1:
         manager.save()
+        for ext_logger in self.loggers:
+          if isinstance(ext_logger, WandbLogger):
+            ext_logger.save_checkpoint(self.model_dir,
+                                      self,
+                                      "train_checkpoints",
+                                      "step",
+                                      current_step,
+                                      max_checkpoints_to_keep,
+                                      checkpoint_on_min=False)
       for c in callbacks:
         c(self, current_step)
       if self.tensorboard and should_log:
         self._log_scalar_to_tensorboard('loss', batch_loss, current_step)
-      if (self.wandb_logger is not None) and should_log:
-        all_data = dict({'train/loss': batch_loss})
-        self.wandb_logger.log_data(all_data, step=current_step)
+      for ext_logger in self.loggers:
+        if isinstance(ext_logger, WandbLogger):
+          ext_logger.log_batch({"loss": batch_loss}, current_step, inputs, labels, group="train")
+        else:
+          ext_logger.log_batch({"loss": batch_loss}, current_step, inputs, labels)
+
 
     # Report final results.
     if averaged_batches > 0:
       avg_loss = float(avg_loss) / averaged_batches
-      logger.info(
+      logs.info(
           'Ending global_step %d: Average loss %g' % (current_step, avg_loss))
       if all_losses is not None:
         all_losses.append(avg_loss)
       last_avg_loss = avg_loss
 
     if checkpoint_interval > 0:
       manager.save()
+      for ext_logger in self.loggers:
+        if isinstance(ext_logger, WandbLogger):
+          ext_logger.save_checkpoint(self.model_dir,
+                                     self,
+                                     "train_checkpoints",
+                                     "step",
+                                     current_step,
+                                     max_checkpoints_to_keep,
+                                     checkpoint_on_min=False)
```

Comment:
The above issues apply here too.  1) There shouldn't be special handling for WandbLogger, and 2) checkpointing by the logger isn't the model's concern.  Logger should define a generic method that gets called at the end of `fit()`.  A subclass can do whatever it wants in that method: save a checkpoint, clean up resources, whatever.